tests/ui/lifetimes/issue-109143.rs

```diff
@@ -0,0 +1,7 @@
+struct Bar<const N: &(for<'a> S<'a>)>;
+//~^ ERROR cannot find trait `S` in this scope
```

Comment:
Could you add a trait named `S` to avoid this off-topic error?

Reply:
Fixed, and move this test to `tests/ui/lifetimes/unusual-rib-combinations.rs`